subprojects/internal-integ-testing/src/main/groovy/org/gradle/test/fixtures/maven/AbstractMavenModule.groovy

```diff
@@ -204,6 +205,13 @@ abstract class AbstractMavenModule extends AbstractModule implements MavenModule
         return this
     }
 
+    /**
+     * Same as {@link #artifact(Map)} since all additional artifacts are undeclared in maven.
+     */
+    MavenModule undeclaredArtifact(Map<String, ?> options) {
```

Comment:
:o: I found the name a bit weird when reading the test. Maybe `extraArtifact` or `arbitraryArtifact`?

Reply:
Mh. _undeclared_ as in "not declared in metadata". I think it's ok.

I actually just copied the name from `IvyFileModule.undeclaredArtifact()`, which exists for years already, to make tests more readable (avoiding an `if ivy ... else ...`).

Changing the name would mean adjusting all tests using it already. ;)

Reply:
Ok no worries, naming is hard :)